### PR TITLE
New nodes

### LINKED ISSUE #869: Implement COUNT DISTINCT
I'm running a query and seems to be having a problem

I'm using source{d} CE and I'm trying to see the unique collaborators for a specific repo.

I'm querying the "commits" table, and my query is:

```sql
SELECT COUNT(DISTINCT commit_author_name), repository_id
FROM commits
WHERE repository_id = 'data-intelligence'
GROUP BY repository_id
```

and I get the number 57.

However, when I run the query:

```sql
SELECT DISTINCT commit_author_name, repository_id
FROM commits
WHERE repository_id = 'data-intelligence'
```

I get the distinct names of 
Marcelo Novaes
Fernanda Gomes
Máximo Cuadros

So, the unique number of collaborators is 3 but I get 57 if I do `COUNT(DISTINCT commit_author_name)`.
I don't understand why.

The version of source{d} CE I'm using is v0.14.0-beta.3

### LINKED ISSUE #921: Update to the new Node API
go-mysql-server Node API changed. We need to apply that changes on gitbase after upgrade to the latest go-mysql-server version.

### PR DESCRIPTION
**Pull request by erizocosmico**
Closes #921
closes #869 

 - [x] I updated the documentation explaining the new behavior if any.
 - [x] I updated CHANGELOG.md file adding the new feature or bug fix.
 - [x] I updated go-mysql-server using `make upgrade` command if applicable.
 - [x] I added or updated examples if applicable.
 - [x] I checked that changes on schema are reflected into the documentation, if applicable.

### CODE REVIEW

**Comment by juanjux** [comment]
Bookmarking this because it's great to see how to use the new API side by side with the old one :)

**Comment by erizocosmico** [comment]
@juanjux 

TL;DR:

```
node.TransformUp(fn)
```

Becomes

```
plan.TransformUp(node, fn)
```

Same with all other transforms and expressions.
